### PR TITLE
Setup sound datagen

### PR DESCRIPTION
**Pull request by Raycoms**
# Changes proposed in this pull request:
- Generates the sounds
- Generate loottable (fix the glassblower one).
-

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/generation/defaults/DefaultSoundProvider.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            map.put(new String[]{"mob.child.female." + soundEvents.name().toLowerCase(Locale.US), "neutral"}, defaultSounds);
+        }
+
+        map.put(new String[]{"mob.barbarian.death", "hostile"}, ImmutableList.of("minecolonies:mob/barbarian/death"));
+        map.put(new String[]{"mob.barbarian.say", "hostile"}, ImmutableList.of("minecolonies:mob/barbarian/say"));
+        map.put(new String[]{"mob.barbarian.hurt", "hostile"}, ImmutableList.of("minecolonies:mob/barbarian/hurt1, minecolonies:mob/barbarian/hurt2, minecolonies:mob/barbarian/hurt3, minecolonies:mob/barbarian/hurt4"));
+
+        map.put(new String[]{"mob.citizen.snore", "neutral"}, ImmutableList.of("minecolonies:mob/citizen/snore"));
+
+        final SoundsJson lootTableJson = new SoundsJson(map);
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/generation/defaults/DefaultSoundProvider.java]: var name^^

File: src/main/java/com/minecolonies/coremod/generation/defaults/DefaultSoundProvider.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        final Path savePath = generator.getOutputFolder().resolve(DataGeneratorConstants.ASSETS_DIR).resolve("sounds.json");
+        IDataProvider.save(DataGeneratorConstants.GSON, cache, lootTableJson.serialize(), savePath);
+    }
+
+    @NotNull
+    @Override
+    public String getName()
+    {
+        return "Default Block Loot Table Provider";
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/generation/defaults/DefaultSoundProvider.java]: here too

File: src/main/generated/resources/data/minecolonies/loot_tables/blocks/blockhutglassblower.json [hunk 3]
```diff
@@ -11,7 +11,7 @@
       "entries": [
         {
           "type": "minecraft:item",
-          "name": "minecolonies:blockhutenchanter",
```
Comment (Asherslab) [inline src/main/generated/resources/data/minecolonies/loot_tables/blocks/blockhutglassblower.json]: can you double check that the enchanter loot_table is still there?

**Review by Asherslab — DISMISSED** [review]
looks good to me

**Review by someaddons — DISMISSED** [review]

**Pull request merged**